crates/proof-of-sql/src/proof_primitive/dynamic_dory_and_hyrax_common_utils/standard_basis_helper.rs

```diff
@@ -1,18 +1,56 @@
-//! This module provides the `build_standard_basis_vecs` method, which is used in converting a point to a
-//! vector used in a Vector-Matrix-Vector product in the dynamic dory scheme.
+use crate::base::scalar::Scalar;
+use alloc::{vec, vec::Vec};
+use core::ops::Mul;
+use itertools::Itertools;
 
-use super::F;
-use ark_ff::{AdditiveGroup, Field};
+/// Compute the hi and lo vectors (or L and R) that are derived from `point`.
+/// L and R are the vectors such that LMR is exactly the evaluation of `a` at the point `point`.
+/// # Panics
+/// This function requires that `point` has length at least as big as the number of rows in `M` that is created by `a`.
+pub(crate) fn compute_dynamic_vecs<
+    S: Scalar,
+    I: Into<S> + From<S> + Copy + core::ops::MulAssign + core::ops::Mul<Output = I>,
+>(
+    point: &[I],
+) -> (Vec<I>, Vec<I>) {
+    let point = point.iter().map(|i| (*i).into());
+    let nu = point.len() / 2 + 1;
+    let mut lo_vec = vec![I::from(S::ZERO); 1 << nu];
+    let mut hi_vec = vec![I::from(S::ZERO); 1 << nu];
+    lo_vec[0] = point
+        .clone()
+        .take(nu)
+        .map(|b| S::ONE - b)
+        .product::<S>()
+        .into();
+    hi_vec[0] = point
+        .clone()
+        .skip(nu)
+        .map(|b| S::ONE - b)
+        .product::<S>()
+        .into();
+    let standard_basis_point: Vec<I> = point
+        .map(|b| I::from((S::ONE - b).inv().expect("Values in point cannot be 1.") - S::ONE))
+        .collect_vec();
+    compute_dynamic_standard_basis_vecs::<S, _>(&standard_basis_point, &mut lo_vec, &mut hi_vec);
+    (lo_vec, hi_vec)
+}
```

Comment:
Please mitigate these clones that this change has introduced:

```suggestion
pub(crate) fn compute_dynamic_vecs<
    S: Scalar,
    I: Into<S> + From<S> + Copy + core::ops::MulAssign + core::ops::Mul<Output = I>,
>(
    point: &[I],
) -> (Vec<I>, Vec<I>) {
    let point: Vec<S> = point.iter().map(|i| (*i).into()).collect();
    let nu = point.len() / 2 + 1;
    let mut lo_vec = vec![I::from(S::ZERO); 1 << nu];
    let mut hi_vec = vec![I::from(S::ZERO); 1 << nu];
    lo_vec[0] = point
        .iter()
        .take(nu)
        .map(|&b| S::ONE - b)
        .product::<S>()
        .into();
    hi_vec[0] = point
        .iter()
        .skip(nu)
        .map(|&b| S::ONE - b)
        .product::<S>()
        .into();
    let standard_basis_point: Vec<I> = point
        .iter()
        .map(|&b| I::from((S::ONE - b).inv().expect("Values in point cannot be 1.") - S::ONE))
        .collect();
    compute_dynamic_standard_basis_vecs::<S, _>(&standard_basis_point, &mut lo_vec, &mut hi_vec);
    (lo_vec, hi_vec)
}


```

Reply:
Further, I'm not sure why you need the I generic here. I think you can remove it entirely from this function as well as the ```compute_dynamic_standard_basis_vecs``` function below. If you really really need it for something, please explain your reasoning, otherwise, i think you can safely drop it. This makes it so that you dont have an empty generic hanging around in your calls to this function, so this:

```rust
compute_dynamic_vecs::<DoryScalar, _>(&point)
compute_dynamic_standard_basis_vecs::<DoryScalar, _>(&point, &mut lo_vec, &mut hi_vec);
```

becomes this:

```rust
compute_dynamic_vecs::<DoryScalar>(&point)
compute_dynamic_standard_basis_vecs::<DoryScalar>(&point, &mut lo_vec, &mut hi_vec);
```

Which is easier to read, maintain, and use in practice. This change will break some of your tests, but all you need to do is remove the second generic from the function call as illustrated above, and then in tests where you have type mismatch errors from this change, you can do something like this:

```rust
    #[test]
    fn we_can_compute_dynamic_vecs_for_length_1_point() {
        let point = vec![F::from(2)];
        let expected_lo_vec = vec![F::from(1), F::from(0)];
        let expected_hi_vec = vec![F::from(1), F::from(1)];
        let point_converted: Vec<DoryScalar> = point.iter().map(|&x: &F| x.into()).collect();
        let (lo_vec, hi_vec) = compute_dynamic_vecs::<DoryScalar>(&point_converted);
        assert_eq!(
            expected_lo_vec
                .iter()
                .map(|&x| x.into())
                .collect::<Vec<DoryScalar>>(),
            lo_vec
        );
        assert_eq!(
            expected_hi_vec
                .iter()
                .map(|&x| x.into())
                .collect::<Vec<DoryScalar>>(),
            hi_vec
        );
    }
```

I would argue this approach is preferred because it is much more readable and explicit about what you are trying to accomplish. If you reallyyyyy need this second I parameter in these two functions, maybe for a future change somewhere else, please document this explicitly, otherwise I think we can safely drop it.

Reply:
That's a good point. None of these functions are externally public, and they get limited use outside unit tests.